sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -118,17 +122,34 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
    * filter should be selective and the filter condition (including expressions in the child
    * plan referenced by the filter condition) should be a simple expression, so that we do
    * not add a subquery that might have an expensive computation.
```

Comment:
let's add more doc to explain why we can extract a sub-plan to filter the application side.
```
The extracted sub-plan should produce a superset of the entire creation side output data, so that
it's still correct to use the sub-plan to build the runtime filter to prune the application side.
```